### PR TITLE
Fix lambda_rank bug for inference.

### LINKED ISSUE #404: 运行lambdarank demo infer阶段代码报错
执行ltr中的lambdarank的infer出现错误，数据为demo中mq2007中的数据。
https://github.com/PaddlePaddle/models/tree/develop/ltr
在lambda_rank.py中infer_data是array，但是到dataprovider_converter.py中dat为一个float型，错误如下，
```
Traceback (most recent call last): File "lambda_rank.py", line 137, in <module> lambda_rank_infer(pass_id=args.num_passes - 1) File "lambda_rank.py", line 120, in lambda_rank_infer output_layer=output, parameters=parameters, input=infer_data) File "/home/yitengfei/python27-gcc482/lib/python2.7/site-packages/paddle/v2/inference.py", line 158, in infer return inferer.infer(field=field, input=input, feeding=feeding) File "/home/yitengfei/python27-gcc482/lib/python2.7/site-packages/paddle/v2/inference.py", line 93, in infer for result in self.iter_infer_field(field=field, **kwargs): File "/home/yitengfei/python27-gcc482/lib/python2.7/site-packages/paddle/v2/inference.py", line 80, in iter_infer_field for result in self.iter_infer(**kwargs): File "/home/yitengfei/python27-gcc482/lib/python2.7/site-packages/paddle/v2/inference.py", line 73, in iter_infer yield self.__gradient_machine__.forwardTest(feeder(data_batch)) File "/home/yitengfei/python27-gcc482/lib/python2.7/site-packages/py_paddle/dataprovider_converter.py", line 284, in __call__ return self.convert(dat, argument) File "/home/yitengfei/python27-gcc482/lib/python2.7/site-packages/paddle/v2/data_feeder.py", line 133, in convert return DataProviderConverter.convert(self, reorder_data(dat), argument) File "/home/yitengfei/python27-gcc482/lib/python2.7/site-packages/py_paddle/dataprovider_converter.py", line 269, in convert scanner.pre_scan(each_step) File "/home/yitengfei/python27-gcc482/lib/python2.7/site-packages/py_paddle/dataprovider_converter.py", line 227, in pre_scan self.__inner_scanner__.pre_scan(each) File "/home/yitengfei/python27-gcc482/lib/python2.7/site-packages/py_paddle/dataprovider_converter.py", line 116, in pre_scan "The input should be a vector, please check your input data." ValueError: The input should be a vector, please check your input data.
```

### PR DESCRIPTION
**Pull request by peterzhang2029**
resolve https://github.com/PaddlePaddle/models/issues/404

### CODE REVIEW

File: ltr/lambda_rank.py [hunk 1]
```diff
@@ -8,25 +8,25 @@
 import paddle.v2 as paddle
 
 
-def lambda_rank(input_dim):
+def lambda_rank(input_dim, is_infer):
     """
-    lambda_rank is a Listwise rank model, the input data and label
+    Lambda_rank is a Listwise rank model, the input data and label
```
Comment (lcy-seso) [inline ltr/lambda_rank.py]: 如果一定要改，改为 LambdaRank，按照论文中的名字。

File: ltr/lambda_rank.py [hunk 2]
```diff
@@ -8,25 +8,25 @@
...
     must be sequences.
 
     https://papers.nips.cc/paper/2971-learning-to-rank-with-nonsmooth-cost-functions.pdf
     parameters :
       input_dim, one document's dense feature vector dimension
 
-    format of the dense_vector_sequence:
+    Format of the dense_vector_sequence:
```
Comment (lcy-seso) [inline ltr/lambda_rank.py]: The format of the dense_vector_sequence is as follows:

File: ltr/lambda_rank.py [hunk 3]
```diff
@@ -44,35 +44,38 @@ def lambda_rank(input_dim):
...
+        # Define cost layer.
+        cost = paddle.layer.lambda_cost(
+            input=output, score=label, NDCG_num=6, max_sort_size=-1)
+        return cost, output
+    return output
 
 
 def train_lambda_rank(num_passes):
-    # listwise input sequence
+    # Listwise input sequence.
```
Comment (lcy-seso) [inline ltr/lambda_rank.py]: 删掉这个注释，或者修改一下，这个注释现在有些不明所以。
The input for LambdaRank is a sequence.

**Review by lcy-seso — COMMENTED** [review]
Almost LGTM.

File: ltr/lambda_rank.py (hunk 1 above)
Comment (peterzhang2029) [reply ltr/lambda_rank.py]: Done

File: ltr/lambda_rank.py (hunk 2 above)
Comment (peterzhang2029) [reply ltr/lambda_rank.py]: Done

File: ltr/lambda_rank.py (hunk 3 above)
Comment (peterzhang2029) [reply ltr/lambda_rank.py]: Done

**Review by lcy-seso — APPROVED** [review]
LGTM.

**Pull request merged**